### PR TITLE
Implements honeypot

### LINKED ISSUE #5: Ban Hosts of Dropped Flows
In case a flow has been banned on port X, such host needs to be banned on all ports (even those not monitored) of the same host. In essence turn this tool into an active honeypot.

### PR DESCRIPTION
**Pull request by frenzis01**
Resolves #5, implementing a honeypot.
A field to specify protected ports has been added to the config file. In case a host sends a packet to one of such ports, it gets banned.
The user can specify (in order of priority) single ports, all ports except (`"!port"`) and port ranges (`"p1-p2"`). These three kinds of data are stored in two maps and a set in `Configuration`.
Now, after checking `isBlacklistedIP(4|6)`, `makeVerdict` checks also whether the host has been ("honey")banned and if so, it checks if it has happened less than 15 minutes ago.
This is done by adding a `Blacklists*` (called `honey_banned`) and a `map` (`honey_banned_time`) to `NwInterface`; there is also another data structure (`std::list honey_banned_timesorted`) needed to perform harvesting, whose head holds the less recent banned host.
Before checking monitored ports, if `dport` is one of the protected ones `src_host` gets added to the `Blacklist` just mentioned. `honey_banned_time` associates to each banned host a timestamp and an iterator to quickly find the host in `honey_banned_timesorted`.

Some notes:
`isProtected(port)` requires _O(log#{ranges})_ to return if it has to look for ranges, only _O(1)_ otherwise.
Harvesting of banned hosts (only up to 10 entries are erased) is performed when there are no packets in the queue.

### CODE REVIEW

**Comment by lucaderi** [comment]
Some comments
- we need to enable the user to specify a list or range of ports, or even something like (all port except)
- blocking users should not be permanent as you did. Better to block them for some time by adding a timeout (e.g. block them for 15 mins). More complicated configurations could be implemented for instance adding exponential blocking of clients caught by the honeypot

This said this work is going in the right direction

File: Configuration.cpp [hunk 1]
```diff
@@ -95,6 +95,19 @@ bool Configuration::readConfigFile(const char *path) {
 	ignored_ports[port] = true;
       }
     }
+
+    all_honeypot_ports = true;
+
+    // Doesn't distinguish between UDP and TCP (and other protocols...)
```
Comment (lucaderi) [inline Configuration.cpp:101]: Your approach works with individual ports, but popular use-cases include
- monitor port ranges
- all ports except.

Example:
- 1024-65535
- !80

**Review by lucaderi — CHANGES_REQUESTED** [review]
Please address this tiny change and make sure it is implemented efficiently with respect to memory and performance.

File: NwInterface.cpp [hunk 2]
```diff
@@ -153,6 +153,9 @@ void NwInterface::packetPollLoop() {
 	break;
       }
     }
+    else { 
+      honeyHarvesting(10);
```
Comment (frenzis01) [inline NwInterface.cpp:157]: should a parameter be added to the config file instead of using _10_?

Comment (lucaderi) [reply NwInterface.cpp:157]: Not that important I would say

**Review by lucaderi — COMMENTED** [review]
Ready to merge

**Pull request merged**